components/ai_chat/renderer/ai_chat_resource_sniffer_url_loader.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_AI_CHAT_RENDERER_AI_CHAT_RESOURCE_SNIFFER_URL_LOADER_H_
+#define BRAVE_COMPONENTS_AI_CHAT_RENDERER_AI_CHAT_RESOURCE_SNIFFER_URL_LOADER_H_
+
+#include <string>
+#include <tuple>
+
+#include "base/memory/scoped_refptr.h"
+#include "base/memory/weak_ptr.h"
+#include "base/task/sequenced_task_runner.h"
+#include "brave/components/ai_chat/renderer/ai_chat_resource_sniffer_throttle_delegate.h"
+#include "brave/components/body_sniffer/body_sniffer_url_loader.h"
+
+namespace body_sniffer {
+class BodySnifferThrottle;
+}  // namespace body_sniffer
+
+namespace ai_chat {
+
+class AIChatResourceSnifferURLLoader
+    : public body_sniffer::BodySnifferURLLoader {
```

Comment:
there is an upcoming refactoring of the body sniffer, hopefully the new sniffer design is more logical than the current

Reply:
sorry i've forgotten to link https://github.com/brave/brave-core/pull/21792/files